OMCompiler/Compiler/NBackEnd/Modules/3_Post/NBSolve.mo

```diff
@@ -595,25 +604,34 @@ public
       derivative := SimplifyExp.simplifyDump(derivative, true, getInstanceName());
 
       if Expression.isZero(derivative) then
-        invertRelation := false;
+        invertRelation := RelationInversion.FALSE;
         status := Status.UNSOLVABLE;
       elseif not Expression.containsCref(derivative, fixed_cref) then
         // If eqn is linear in cref:
         (eqn, funcTree) := solveLinear(eqn, residual, derivative, diffArgs, fixed_cref, funcTree);
         // If the derivative is negative, invert possible inequality sign
-        invertRelation := Expression.isNegative(derivative);
+        invertRelation := if Expression.isNegative(derivative) then RelationInversion.FALSE else RelationInversion.TRUE;
```

Comment:
If `isNegative` returns `false` we don't know if `derivative` is non-negative, so you should also test if it is positive and if it's neither return unknown.
```suggestion
        invertRelation := if Expression.isPositive(derivative) then RelationInversion.TRUE else if Expression.isNegative(derivative) then RelationInversion.FALSE else RelationInversion.UNKNOWN;
```